compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -3555,6 +3556,13 @@ public void visit(BLangInvocation iExpr) {
 
         if (iExpr.symbol != null && iExpr.symbol.kind == SymbolKind.ERROR_CONSTRUCTOR) {
             result = rewriteErrorConstructor(iExpr);
+        } else if (iExpr.symbol.kind == SymbolKind.FUNCTIONAL_CONSTRUCTOR) {
+            String name = ((BConstructorSymbol) iExpr.symbol).name.value;
+
+            String internalMethodName = name.substring(0, 1).toLowerCase() + name.substring(1) + "Ctor";
```

Comment:
Cant we avoid this extra string processing by having the same name for the ctor? Or at least reduce some of the processing. eg: Can we avoid "Ctor" part?

Reply:
In my view having a clear name describing what it does far outweigh the cost of string concat/substring. Especially given that this is not in a hot path.